lib/hawkular/inventory/entities.rb

```diff
@@ -0,0 +1,217 @@
+# It contains class definitions that are used by the inventory REST client
+module Hawkular::Inventory
+  # A Basic inventory entity with id, name, path and optional properties
+  class BaseEntity
+    # @return [String] Full path of the entity
+    attr_reader :path
+    # @return [String] Name of the entity
+    attr_reader :name
+    # @return [String] Name of the entity
+    attr_reader :id
+    # @return [String] Feed this entity belongs to (or nil in case of a feedless entity)
+    attr_reader :feed
+    # @return [String] Name of the environment for this entity
+    attr_reader :env
+    # @return [String] Properties of this entity
+    attr_reader :properties
+
+    def initialize(hash)
+      @id = hash['id']
+      @path = hash['path']
+      @name = hash['name'] || @id
+      @properties = hash['properties'] || {}
+      @_hash = hash.dup
+
+      return if @path.nil?
+
+      tmp = @path.split('/')
+      tmp.each do |pair|
+        (key, val) = pair.split(';')
+        case key
+        when 'f'
+          @feed = val
+        when 'e'
+          @env = val
+        end
+      end
+    end
+
+    def ==(other)
+      self.eql?(other) || other.class == self.class && other.id == @id
+    end
+
+    # Returns a hash representation of the resource type
+    # @return [Hash<String,Object>] hash of the type
+    def to_h
+      @_hash.dup
+    end
+  end
+
+  # A ResourceType is like a class definition for {Resource}s
+  # ResourceTypes are currently unique per feed, but one can assume
+  # that a two types with the same name of two different feeds are
+  # (more or less) the same.
+  class ResourceType < BaseEntity
+    def initialize(rt_hash)
+      super(rt_hash)
+    end
+  end
+
+  # A Resource is an instantiation of a {ResourceType}
+  class Resource < BaseEntity
+    # @return [String] Full path of the {ResourceType}
+    attr_reader :type_path
+
+    def initialize(res_hash)
+      super(res_hash)
+      if res_hash.key? :resourceTypePath
+        @type_path = res_hash[:resourceTypePath]
+      else
+        @type = res_hash['type']
+        @type_path = res_hash['type']['path']
+      end
+    end
+  end
+
+  class MetricType < BaseEntity
+    # @return [String] GAUGE, COUNTER, etc.
+    attr_reader :type
+    # @return [String] metric unit such as NONE, BYTES, etc.
+    attr_reader :unit
+    # @return [Long] collection interval in seconds
+    attr_reader :collection_interval
+
+    def initialize(type_hash)
+      super(type_hash)
+      @type = type_hash['type']
+      @unit = type_hash['unit']
+      @collection_interval = type_hash['collectionInterval']
+    end
+  end
+
+  # Definition of a Metric inside the inventory.
+  class Metric < BaseEntity
```

Comment:
Hmm, it's tempting to do `Metric < MetricType` and in the Metric constructor call `super(metric_hash['type'])`.

But do we want to have this `Metric  --- is_a ----> MetricType` relation? It may be dangerous, what if api consumers calls those tests? Introducing a new (private) class and have both Metric and MetricType extend it. Hmm, imho doesn't worth doing it, but I can do that if you insist.

Btw. same holds for `Resource` and `ResourceType` and again, this wasn't part of this PR.